src/install/lockfile/Package/Meta.zig

```diff
@@ -28,12 +29,24 @@ pub const Meta = extern struct {
         true,
     } = .false,
 
-    _padding_integrity: [2]u8 = .{0} ** 2,
+    _padding_integrity: u8 = 0,
+    _padding_end: u8 = 0,
 
     /// Does the `cpu` arch and `os` match the requirements listed in the package?
     /// This is completely unrelated to "devDependencies", "peerDependencies", "optionalDependencies" etc
     pub fn isDisabled(this: *const Meta) bool {
-        return !this.arch.isMatch() or !this.os.isMatch();
+        return !this.arch.isMatch() or !this.os.isMatch() or !this.libc.isMatch();
+    }
+
+    pub fn isDisabledWithTarget(this: *const Meta, target_os: ?Npm.OperatingSystem, target_cpu: ?Npm.Architecture, target_libc: ?Npm.Libc) bool {
```

Comment:
this function looks like a bug. shouldn't all `isDisabled` calls check the manager.options.target_cpu/os/libc config?

Reply:
I believe the seperate function is to have input of current instead using system. I didn't think the current api for that should be changed, so new fn made sense. But I could modify existing and do an orelse maybe.